### ISSUE TITLE
[BUG] Rspec running infinitely

### ISSUE THREAD

**Issue by DavidsonGM** [excid3/noticed#371]
## Bug Report

**Describe the Bug:**

I have started using this gem recently, so please correct me if I have done something wrong and if I forgot to provide any information. I'm currently facing the following issue: 

Sometimes when delivering notifications trough callbacks (such as `after_create` and `after_update`) the specs never finish running and I can't really figure out why. Despite of that, everithing seems to be working fine in development mode.

I guess the problem might be related to the `deliver` method, since the only workaround I found was setting the `wait` option in the method. Since i setted it to `5.seconds` in every single call to this method, I didn't see this error again. So it's not a major issue for me, but I think it would be nice if it could be fixed.

**To Reproduce:**
I am not sure what caused this, so I will put here a sample of my code in which this error happens 

1. I have created a default notifier class, which is parent of all other notifier classes
  ```rb
  # app/notifiers/default_notifier.rb

  class DefaultNotifier < Noticed::Event
    deliver_by :action_cable do |config|
      config.channel = "NotificationsChannel"
      config.stream = -> { recipient.id }
      config.message = -> { NotificationSerializer.new(self).as_json }
    end
  end

  ```
2. Message notifier class
```rb
  # app/notifiers/message_notifier.rb

class MessageNotifier < DefaultNotifier
  notification_methods do
    def name
      "Mensagem recebida de #{record.user.name}"
    end

    def message
      record.body
    end

    def url
      chats_path(id: record.chat.id)
    end
  end
end

``` 
3. Message callback
```rb
# app/models/message.rb
  after_create :notify_receiver

  private

  def notify_receiver
    MessageNotifier.with(record: self).deliver(receiver)
  end
```

4. Spec file (we are using factory bot, in case this matters)
```rb
# spec/models/message_spec.rb
require "rails_helper"

RSpec.describe Message, type: :model do
  describe "default factory" do
    it { expect(create(:message)).to be_valid }
  end

  describe "validations" do
    context "when user does not belongs to chat" do
      let(:user) { create(:user) }
      let(:chat) { create(:chat) }

      it { expect(build(:message, chat:, user:)).not_to be_valid }
    end

    context "with empty body" do
      it { expect(build(:message, body: "")).not_to be_valid }
    end
  end
end
```

5. When running ` rspec spec/models`, this tests don't exactly fail, but the other tests never get to run.

**Expected Behavior:**

All the tests finish running (even if it's with a fail message)

**Actual Behavior:**

Rspec stuck in an infinite loop.

**Screenshots (if applicable):**

**Environment:**
- Noticed gem version: 2.0.1
- Ruby version: 3.2.2
- Rails version: 7.1.1
- Operating System: Zorin OS 16.3 (But also fails on our github CI, which runs on ubuntu 22.04.3) 

**Additional Context:**

**Possible Fix:**

**Steps to Reproduce with Fix (if available):**

**Related Issues:**

I've seen the issue  #100 which seems to be the very same problem, but it couldn't find out if it has been fixed yet. 

**Labels to Apply:**

**Checklist:**

- [x] I have searched for similar issues and couldn't find any
- [x] I have checked the documentation for relevant information
- [x] I have included all the required information

**Comment by excid3** [comment]
What happens if you use this instead?

```ruby
 after_create_commit :notify_receiver
```

**Comment by excid3** [comment]
Tested this in our test suite with minitest and it works fine.

```ruby
class Comment < ApplicationRecord
  after_create do
    CommentNotifier.with(record: self).deliver(User.first)
  end
end
```

Doesn't seem like it's a bug in Noticed and is more likely an issue with your test suite somewhere. Could be FactoryBot related?

**Comment by DavidsonGM** [comment]
At first, I was using the `after_create_commit` callback. Changed it to `after_create` to see if it could be the problem but decided to keep it.
I tried to refactor one of the tests which caused the issue not to use the factory bot methods, but it didn't fix it. So I can't confirm if the problem is related to FactoryBot.

Also, I tried running the tests again today and the RSpec was no longer running forever. Instead, the following error is raised on the test right after a test which called a create.
```
 ActiveRecord::StatementInvalid:
            PG::UnableToSend: SSL SYSCALL error: EOF detected
            SSL error: sslv3 alert bad record mac
```

After this, all the following tests fails with the message
```
Failure/Error: result = conn.async_exec(sql)
          
          ActiveRecord::StatementInvalid:
            PG::UnableToSend: no connection to the server
```

Since the workaround I found seems to be working fine, I probably won't dig much more into this. But I still wanted to let you guys know this issue existed. I'm not very familiar with de development  rails gems, but let me know if I could help with any additional info that could be useful or in any other way.
